src/contexts/UserContext.tsx

```diff
@@ -0,0 +1,82 @@
+import React, {
+  createContext,
+  Dispatch,
+  SetStateAction,
+  useState,
+} from 'react';
+
+interface UserState {
+  userState: {
+    _deleted: null;
+    _lastChangedAt: 0;
+    _version: 1;
+    createdAt: '';
+    displayName: '';
+    homes: {
+      items: [];
+      nextToken: null;
+      startedAt: null;
+    };
+    id: '';
+    owner: '';
+    updatedAt: '';
+    username: '';
+  };
```

Comment:
Can we not use the type `User` from `API.ts`?

Reply:
I've described why homeInfo exists below.